packages/@ember/-internals/routing/lib/services/router.ts

```diff
@@ -61,19 +65,30 @@ function cleanURL(url: string, rootURL: string) {
    @class RouterService
  */
 export default class RouterService extends Service {
-  _router!: EmberRouter;
+  get _router(): EmberRouter {
+    let router = this[ROUTER];
+    if (router !== undefined) {
+      return router;
+    }
+    const owner = getOwner(this) as any;
+    router = owner.lookup('router:main');
+    router.setupRouter();
+    return (this[ROUTER] = router);
+  }
+
+  constructor(owner: Owner) {
+    super(owner);
 
-  init() {
-    super.init(...arguments);
+    const router = owner.lookup('router:main') as EmberRouter;
```

Comment:
Just to confirm, we have to `.lookup` here because `owner.router` may not be setup?

Reply:
Made the change because in the added `RenderingTestCase` there's no `owner.router`, and `router` is also not on `Owner` type.